packages/flutter/lib/src/cupertino/radio.dart

```diff
@@ -377,27 +439,56 @@ class _RadioPainter extends ToggleablePainter {
         canvas.drawPath(path, checkPaint);
       }
     } else {
-      // Outer border
-      canvas.drawCircle(center, _kOuterRadius, paint);
-
-      paint.style = PaintingStyle.stroke;
-      paint.color = CupertinoColors.inactiveGray;
-      canvas.drawCircle(center, _kOuterRadius, paint);
-
       if (value ?? false) {
-        paint.style = PaintingStyle.fill;
-        paint.color = activeColor;
-        canvas.drawCircle(center, _kOuterRadius, paint);
-        paint.color = fillColor;
-        canvas.drawCircle(center, _kInnerRadius, paint);
+        final Paint outerPaint = Paint()
+        // In dark mode, the outer color of an active radio is slightly darker.
+          ..color = isActive
+            ? (isDefaultSelected && brightness == Brightness.dark ? _kDarkModeOuterColor : activeColor)
```

Comment:
The logic around `isDefaultSelected` feels pretty weird to me. For example, if the `fillColor` is customized, then suddenly the active outer color in dark mode changes from `_kDarkModeOuterColor` to `activeColor`, which is very surprising.

Is there a reason why a more solid way isn't feasible? For example, can we make the active color a `CupertinoDynamicColor`?

Reply:
Yes. That's very true. I've refactored to do this outside the painter instead, using `CupertinoDynamicColor` to resolve the default dark mode color. Thanks for pointing this out.